swagger/src/test/scala/org/http4s/rho/swagger/TypeBuilderSuite.scala

```diff
@@ -0,0 +1,540 @@
+  type Foos = Seq[Foo]
+  case class FooComposite(single: Foo, many: Seq[Foo])
+  case class FooCompositeWithAlias(single: Bar, many: Seq[Bar], manyAlias: Foos)
+  case class FooWithListOfLists(values: Seq[Seq[Int]])
+  case class FooWithList(l: List[Int])
+  case class FooWithMap(l: Map[String, Int])
+  case class FooVal(foo: Foo) extends AnyVal
+  case class BarWithFooVal(fooVal: FooVal)
+  @DiscriminatorField("foobar")
+  sealed trait Sealed {
+    def foo: String
+  }
+  case class FooSealed(a: Int, foo: String, foo2: Foo) extends Sealed
+  case class BarSealed(str: String, foo: String) extends Sealed
+
+  sealed trait SealedEnum
+  case object FooEnum extends SealedEnum
+  case object BarEnum extends SealedEnum
+
+  case class SealedEnumContainer(e: SealedEnum)
+
+  sealed trait BTree
+  case class Branch(left: BTree, right: BTree) extends BTree
+  case class Leaf(value: String) extends BTree
+
+  case class BarList(tail: Option[BarList])
+
+  case class FooEither(stringOrFoo: Either[String, FooEither])
+
+  sealed trait TopLevelSealedTrait
+  case class Value1() extends TopLevelSealedTrait
+  sealed trait LowerLevelSealedTrait extends TopLevelSealedTrait
+  case class Value2() extends LowerLevelSealedTrait
+  case class Value3() extends LowerLevelSealedTrait
+
+  sealed trait MixedSealed
+  case class CaseClass() extends MixedSealed
+  case object CaseObject extends MixedSealed
+
+  trait Outer[T] {
+    case class Inner(t: T)
+  }
+  object OuterInt extends Outer[Int]
+
+  type ShapelessIntOrString = Int :+: String :+: CNil
+
+  def modelOf[T](implicit t: TypeTag[T], st: ShowType): Set[Model] =
+    modelOfWithFormats(DefaultSwaggerFormats)
+
+  def modelOfWithFormats[T](
+      formats: SwaggerFormats)(implicit t: TypeTag[T], st: ShowType): Set[Model] =
+    TypeBuilder.collectModels(t.tpe, Set.empty, formats, typeOf[IO[_]])
+}
+
+class TypeBuilderSuite extends FunSuite {
+  import model._
+  import models.{ArrayProperty, Model, RefProperty}
+
+  implicit val showType: ShowType = DefaultShowType
+
+  test("A TypeBuilder should not Build a model for a primitive") {
+    val primitives = Set[TypeTag[_]](
+      typeTag[String],
+      typeTag[Int],
+      typeTag[Long],
+      typeTag[Double],
+      typeTag[Float],
+      typeTag[Byte],
+      typeTag[BigInt],
+      typeTag[Boolean],
+      typeTag[Short],
+      typeTag[java.lang.Integer],
+      typeTag[java.lang.Long],
+      typeTag[java.lang.Double],
+      typeTag[java.lang.Float],
+      typeTag[BigDecimal],
+      typeTag[java.lang.Byte],
+      typeTag[java.lang.Boolean],
+      typeTag[Number],
+      typeTag[java.lang.Short],
+      typeTag[Date],
+      typeTag[Timestamp],
+      typeTag[Symbol],
+      typeTag[java.math.BigDecimal],
+      typeTag[java.math.BigInteger]
+    )
+
+    val ms = primitives.foldLeft(Set.empty[Model]) { (s, t) =>
+      TypeBuilder.collectModels(t.tpe, s, DefaultSwaggerFormats, typeOf[IO[_]])
+    }
+
+    assert(ms.isEmpty)
```

Comment:
Could we refactor this (and maybe other `.isEmpty` tests so that they `assertEquals(ms, List.empty)` (roughly speaking).  This way when the assertion fails we can see what the values are that were not supposed to be there. 
 I would also be open to adding some additional `clue`s.

Reply:
Good point, I've fixed that.